docs/linux/fs.md

```diff
@@ -6,127 +6,135 @@ sidebar_label: File Systems
 persistently stores a magic number and optional feature flags and values. On
 mount, *zonefs* uses the block layer API function `blkdev_report_zones()` to
 obtain the device zone configuration and populates the mount point with a
-static file tree solely based on this information. File sizes come from the
-device zone type and write pointer position managed by the device itself.
+static file tree that is based solely on this information. File sizes come
+from the device zone type and the write-pointer position is managed by the
+device itself.
 
 The super block is always written on disk at sector 0. The first zone of the
-device storing the super block is never exposed as a zone file by *zonefs*. If
-the zone containing the super block is a sequential zone, the `mkzonefs` format
-tool always "finishes" the zone, that is, it transitions the zone to a full
-state to make it read-only, preventing any data write.
+device that stores the super block is never exposed as a zone file by
+*zonefs*. If the zone that contains the super block is a sequential zone, the
+`mkzonefs` format tool always "finishes" the zone (that is, it transitions the
+zone to a full state to make it read-only, preventing any data write).
 
 ### Zone Type Sub-Directories
 
-Files representing zones of the same type are grouped together under the same
-sub-directory automatically created on mount.
+Files that represent zones of the same type are grouped together under the same
+sub-directory, which is automatically created on mount.
 
 For conventional zones, the sub-directory "cnv" is used. This directory is
-however created if and only if the device has usable conventional zones. If
-the device only has a single conventional zone at sector 0, the zone will not
-be exposed as a file as it will be used to store the *zonefs* super block. For
-such devices, the "cnv" sub-directory will not be created.
+created only if the device has usable conventional zones. If the device has
+only a single conventional zone at sector 0, the zone will not be exposed as a
+file (because it will be used to store the *zonefs* super block). For such
+devices, the "cnv" sub-directory will not be created.
 
 For sequential write zones, the sub-directory "seq" is used.
 
 These two directories are the only directories that exist in *zonefs*. Users
-cannot create other directories and cannot rename nor delete the "cnv" and
-"seq" sub-directories.
+cannot create other directories and can neither rename nor delete the "cnv"
+and "seq" sub-directories.
 
-The size of the directories indicated by the `st_size` field of `struct stat`,
-obtained with the `stat()` or `fstat()` system calls, indicates the number of
-files existing under the directory.
+The size of the directories is indicated by the `st_size` field of `struct
+stat`, which is obtained with the `stat()` or `fstat()` system calls,
+indicates the number of files that exist under the directory.
```

Comment:
2 verbs in the sentence: ...is indicated...indicates the number.
May be something like this is better:

The size of the directories indicates the number of files that exist under the directory. This size is indicated by the `st_size` field of `struct stat`, which is obtained with the `stat()` or `fstat()` system calls.

Reply:
Excellent. Altered just so.